core/drivers/pl011.c

```diff
@@ -108,6 +108,13 @@ static void pl011_flush(struct serial_chip *chip)
 {
 	vaddr_t base = chip_to_base(chip);
 
+	/*
+	 * It can happen that Linux initializes the OP-TEE driver with the
+	 * console UART disabled. Avoid an infinite loop in such a case.
+	 */
+	if (!(read32(base + UART_CR) & UART_CR_UARTEN))
+		return;
+
 	while (!(read32(base + UART_FR) & UART_FR_TXFE))
 		;
```

Comment:
What if the uart is disabled while in this loop?

Reply:
Then it will hang. Would you rather move the CR read inside the loop? This is not required to deal with the case I observed, but it would obviously make the code even more robust.